File: win-prog/ProgWinVB/Essential Structures/HelloCenteredAlignment/HelloCenteredAlignment.vb
'-------------------------------------------------------
' HelloCenteredAlignment.vb (c) 2002 by Charles Petzold
'-------------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class HelloCenteredAlignment
    Inherits Form

    Shared Sub Main()
        Application.Run(New HelloCenteredAlignment())
    End Sub

    Sub New()
        Text = "Hello Centered Using String Alignment"
        BackColor = SystemColors.Window
        ForeColor = SystemColors.WindowText
        ResizeRedraw = True
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Dim grfx As Graphics = pea.Graphics
        Dim strfmt As New StringFormat()

        strfmt.Alignment = StringAlignment.Center
        strfmt.LineAlignment = StringAlignment.Center

        grfx.DrawString("Hello, world!", Font, New SolidBrush(ForeColor), _
                        ClientSize.Width \ 2, ClientSize.Height \ 2, strfmt)
    End Sub
End Class

File: win-prog/ProgWinCS/Essential Structures/HelloCenteredAlignment/HelloCenteredAlignment.cs
//-----------------------------------------------------
// HelloCenteredAlignment.cs © 2001 by Charles Petzold
//-----------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class HelloCenteredAlignment: Form
{
     public static void Main()
     {
          Application.Run(new HelloCenteredAlignment());
     }
     public HelloCenteredAlignment()
     {
          Text = "Hello Centered Using String Alignment";
          BackColor = SystemColors.Window;
          ForeColor = SystemColors.WindowText;
          ResizeRedraw = true;
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Graphics     grfx   = pea.Graphics;
          StringFormat strfmt = new StringFormat();

          strfmt.Alignment     = StringAlignment.Center;
          strfmt.LineAlignment = StringAlignment.Center;

          grfx.DrawString("Hello, world!", Font, new SolidBrush(ForeColor), 
                          ClientSize.Width / 2, ClientSize.Height / 2, 
			              strfmt);
     }
}